calculator/src/App.js

```diff
@@ -0,0 +1,173 @@
+import React from "react";
+import "./App.css";
+
+class App extends React.Component {
+  state = { input: "", previousNum: "", currentNum: "", operator: "" };
+
+  handleClick = (e) => {
+    this.setState({
+      input: this.state.input + e.target.value,
+    });
+  };
+  // clears input when clicked
+  handleClear = (e) => {
+    this.setState({ input: "" });
+  };
+
+  handlePlusNegative = (e) => {
+    this.setState({
+      input: Number(this.state.input) * -1,
+    });
+  };
+
+  add = () => {
+    this.setState({
+      previousNum: this.state.input,
+      input: "",
+      operator: "plus",
+    });
+  };
+
+  handleEqual = () => {
+    // console.log(this.state.input);
+    this.state.currentNum = this.state.input;
+    if (this.state.input === "") {
+      // console.log("its zero");
+      this.setState({
+        input: 0,
+        previousNum: "",
+        currentNum: "",
+        operator: "",
+      });
+    } else if (this.state.operator === "plus") {
```

Comment:
Restructuring out the operator from state would cut down on repetition

A switch statement would be better suited for this conditional. 
switch(operation) {
 case: "+":